core/services/configuration_generator.rb

```diff
@@ -1,16 +1,251 @@
 # frozen_string_literal: true
 
-require_relative '../models/result'
 require_relative 'product_attributes'
+require_relative '../out'
+require_relative '../models/result'
 require_relative '../models/tool_configuration'
 
-# The class provides methods for generating the role of the file.
-module ConfigurationGenerator
+# The class provides methods for generating attributes for configuration generator
+class ConfigurationGenerator
+  def initialize(log, env)
+    @ui = log
+    @env = env
+  end
+
+  # Make product config and recipe name for install it to the VM.
+  #
+  # @param product [Hash] parameters of product to configure from configuration file
+  # @param box [String] name of the box
+  # @return [Result::Base] recipe name and product config in format { recipe: String, config: Hash }.
+  # rubocop:disable Metrics/MethodLength
+  def make_product_config_and_recipe_name(product, box)
+    repo = nil
+    if !product['repo'].nil?
+      repo_name = product['repo']
+      @ui.info("Repo name: #{repo_name}")
+      unless @env.repos.knownRepo?(repo_name)
```

Comment:
Get all services from `@env` in the constructor. Do not store env in the constructor as class instance variable.